application/dialogs/AccountLoginDialog.cpp

```diff
@@ -0,0 +1,127 @@
+// Licensed under the Apache-2.0 license. See README.md for details.
+
+#include "AccountLoginDialog.h"
+#include "ui_AccountLoginDialog.h"
+
+#include "auth/AccountModel.h"
+#include "auth/BaseAccount.h"
+#include "auth/BaseAccountType.h"
+#include "tasks/Task.h"
+#include "resources/ResourceProxyModel.h"
+#include "MultiMC.h"
+
+AccountLoginDialog::AccountLoginDialog(QWidget *parent) :
+	QDialog(parent),
+	ui(new Ui::AccountLoginDialog)
+{
+	setWindowTitle(tr("New Account"));
+	ui->setupUi(this);
+	ui->progressWidget->setVisible(false);
+	ui->errorLbl->setVisible(false);
+	ui->loginBtn->setFocus();
+	ui->usernameEdit->setFocus();
+	ui->typeBox->setModel(ResourceProxyModel::mixin<QIcon>(MMC->accountsModel()->typesModel()));
+
+	connect(ui->typeBox, static_cast<void(QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &AccountLoginDialog::currentTypeChanged);
+	currentTypeChanged(0);
+}
+AccountLoginDialog::AccountLoginDialog(const QString &type, QWidget *parent)
+	: AccountLoginDialog(parent)
+{
+	setWindowTitle(tr("New Account for %1").arg(MMC->accountsModel()->type(type)->text()));
+	ui->typeBox->setCurrentIndex(ui->typeBox->findData(type));
+	ui->typeBox->setDisabled(true);
+
+	setupForType(type);
+}
+AccountLoginDialog::AccountLoginDialog(BaseAccount *account, QWidget *parent)
+	: AccountLoginDialog(account->type(), parent)
+{
+	setWindowTitle(tr("Login"));
+	m_account = account;
+	ui->usernameEdit->setText(account->loginUsername());
+}
+
+AccountLoginDialog::~AccountLoginDialog()
+{
+	delete ui;
+}
+
+void AccountLoginDialog::setSession(SessionPtr session)
+{
+	m_session = session;
+}
+
+void AccountLoginDialog::on_cancelBtn_clicked()
+{
+	reject();
```

Comment:
NOTE: reject is a REJECT. By the USER. You don't go asking again in an infinite loop.